s4e-web/src/app/views/map-view/view-manager/view-manager.component.ts

```diff
@@ -45,49 +45,37 @@ export class ViewManagerComponent implements OnInit, OnDestroy {
     }
   }
 
-  loading$: Observable<boolean>;
   scenes: IUILayer[] = [];
-  products$: Observable<IUILayer[]>;
-  productsLoading$: Observable<boolean>;
-  overlays$: Observable<IUILayer[]>;
-  overlaysLoading$: Observable<boolean>;
-  searchResults$: Observable<LocationSearchResult[]>;
-  searchResultsLoading$: Observable<boolean>;
+  groupedProducts$: Observable<IUILayer[][]> = this.productQuery.selectGroupedProducts();
+  productsLoading$: Observable<boolean> = this.productQuery.selectLoading();
+  overlays$: Observable<IUILayer[]> = this.overlayQuery.selectVisibleAsUIOverlays();
+  overlaysLoading$: Observable<boolean> = this.overlayQuery.selectLoading();
+  loading$: Observable<boolean> = combineLatest([this.overlaysLoading$, this.productsLoading$])
+    .pipe(map(([overlayLoading, productsLoading]) => overlayLoading || productsLoading));
```

Comment:
We have a couple of basic operators which can handle this kind of mappings, the one you can use is `mapIsAnyTrue()`